30413/Alisa Dariana Roman/Lab05/extra/main.c

```diff
@@ -0,0 +1,107 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/**
+first i observe that the operators are only on positions that represent powers of 2
+also, the operands are always leafs
+*  -  1.05  +  -55  22  10.3
+1  2   3    4   5   6     7
+
+algorithm:
+- create binary tree with operators, when an operand is met stop there with that branch, operand remains leaf
+**/
+
+typedef struct node_type
+{
+    char item[10];
+    struct node_type *left, *right;
+} NodeT;
+
+int isOperator(char item[]) /// it's an operator if the size of <item[]> is 1 and is +, -, * or /
+{
+    if(strlen(item) == 1) {
+        char c = item[0];
+        if (c == '+') {
+            return 1;
+        }
+        if (c == '-') {
+            return 1;
+        }
+        if (c == '*') {
+            return 1;
+        }
+        if (c == '/') {
+            return 1;
+        }
+        return 0;
+    }
+    return 0;   /// return 0 if size is greater than 1
+}
+
+NodeT *createNode(char item[])
+{
+    NodeT *p = (NodeT*) malloc (sizeof(NodeT));
+    if (p) {
+        strcpy(p->item, item);
+        p->left = p->right = NULL;
+    }
+    return p;
+}
+
+NodeT *createBinTree(FILE *fileI)
```

Comment:
e in regula?

Reply:
Buna rezolvare, felicitari! Ai ales varianta recursiva care se foloseste de stack-ul programului asa ca nu este nevoie sa iti definesti tu propriul stack.